From 564c6a7564b81d2d44d610655140dda2516b51a7 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 2 Apr 2021 07:10:21 +0000
Subject: [PATCH] GCC: fix type confusion in AdsPageLoadMetricsObserver

GCC gets confused if an alias is declared with name of another
type, e.g. "using AggregateFrameData = AggregateFrameData".
This can be avoided by adding explicit namespace.
---
 .../observers/ad_metrics/ads_page_load_metrics_observer.h   | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/components/page_load_metrics/browser/observers/ad_metrics/ads_page_load_metrics_observer.h b/components/page_load_metrics/browser/observers/ad_metrics/ads_page_load_metrics_observer.h
index 06db558..fa4fbe9 100644
--- a/components/page_load_metrics/browser/observers/ad_metrics/ads_page_load_metrics_observer.h
+++ b/components/page_load_metrics/browser/observers/ad_metrics/ads_page_load_metrics_observer.h
@@ -47,9 +47,9 @@ class AdsPageLoadMetricsObserver
     : public PageLoadMetricsObserver,
       public subresource_filter::SubresourceFilterObserver {
  public:
-  using AggregateFrameData = AggregateFrameData;
-  using FrameTreeData = FrameTreeData;
-  using ResourceMimeType = ResourceMimeType;
+  using AggregateFrameData = page_load_metrics::AggregateFrameData;
+  using FrameTreeData = page_load_metrics::FrameTreeData;
+  using ResourceMimeType = page_load_metrics::ResourceMimeType;
   using ApplicationLocaleGetter = base::RepeatingCallback<std::string()>;
 
   // Helper class that generates a random amount of noise to apply to thresholds
-- 
2.26.3

